WalletWasabi/Crypto/Slip21Node.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Linq;
+using System.Text;
+using NBitcoin;
+using static NBitcoin.Crypto.Hashes;
+
+namespace WalletWasabi.Crypto
+{
+	public class Slip21Node
+	{
+		private static readonly int KEY_SIZE = 32;
+		private byte[] _data;
+
+		public Slip21Node(byte[] data) =>
+			_data = data.Length == 64
+				? data
+				: throw new ArgumentException("The data array has to be 64 bytes long.", nameof(data));
```

Comment:
What do you think of using `NBitcoin.ExtKey` instead of `byte[]`, and then relying on the `Key` property instead of `KEY_SIZE`?

Reply:
Do you mean using the `ExtKey.Key` instead of `data[..32]` and `ExtKey.ChainCode` instead of `data[32..]`?

Reply:
I did, but I misread the code (sorry I've been trying to catch up with too many things and was sloppy/hasty) and thought the data being deconstructed is the BIP32 master seed instead of the output of the SLIP-21 derivation.

I think my suggestion is silly when looking at this PR, since `DeriveSeed()` returns `byte[]`, whereas I thought it returned the equivalent of `ExtKey(mnemonic.DeriveSeed())`.

However, in general I would still be more comfortable with relying on passing around typed objects instead of bytestrings, and i think it does make sense in the context of `KeyManager` since `GetMasterExtKey` returns an `ExtKey` and is likely what will be used given that Wasabi keeps it encrypted